File: src/main/java/com/hospital/management/department/service/DepartmentService.java
package com.hospital.management.department.service;

import com.hospital.management.department.dto.DepartmentRequestDTO;
import com.hospital.management.department.dto.DepartmentResponseDTO;

import java.util.List;

public interface DepartmentService {

    DepartmentResponseDTO saveDepartment(DepartmentRequestDTO dto);

    List<DepartmentResponseDTO> getAllDepartments();

    DepartmentResponseDTO getDepartmentById(Long id);

    DepartmentResponseDTO updateDepartment(Long id, DepartmentRequestDTO dto);

    void deleteDepartment(Long id);
}

File: src/services/departmentService.js
import api from "./api";

// GET ALL DEPARTMENTS
export const getAllDepartments = () => {
  return api.get("/api/departments");
};

// GET DEPARTMENT BY ID
export const getDepartmentById = (id) => {
  return api.get(`/api/departments/${id}`);
};

// SAVE DEPARTMENT
export const saveDepartment = (department) => {
  return api.post("/api/departments", department);
};

// UPDATE DEPARTMENT
export const updateDepartment = (id, department) => {
  return api.put(`/api/departments/${id}`, department);
};

// DELETE DEPARTMENT
export const deleteDepartment = (id) => {
  return api.delete(`/api/departments/${id}`);
};
export const getAllDepartments = () => {
  return api.get("/api/departments");
};